src/main/java/mate/academy/springintro/service/impl/OrderServiceImpl.java

```diff
@@ -0,0 +1,42 @@
+package mate.academy.springintro.service.impl;
+
+import jakarta.persistence.EntityNotFoundException;
+import java.util.List;
+import java.util.Set;
+import lombok.RequiredArgsConstructor;
+import mate.academy.springintro.config.OrderMapper;
+import mate.academy.springintro.dto.order.OrderRequestDto;
+import mate.academy.springintro.model.Order;
+import mate.academy.springintro.model.OrderItem;
+import mate.academy.springintro.model.OrderStatus;
+import mate.academy.springintro.repository.OrderRepository;
+import mate.academy.springintro.service.OrderService;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+
+@RequiredArgsConstructor
+@Service
+public class OrderServiceImpl implements OrderService {
+    private final OrderRepository orderRepository;
+    private final OrderMapper orderMapper;
+
+    @Override
+    public Order placeOrder(Long userId, String shippingAddress, Set<OrderItem> orderItems) {
+        OrderRequestDto orderRequestDto = new OrderRequestDto(shippingAddress, orderItems);
+        Order order = orderMapper.toOrder(userId, orderRequestDto, orderItems);
+        return orderRepository.save(order);
```

Comment:
This method should be completely rewritten, let's discuss it in the chat or Q&A session if you will have questions